services/spar/src/Spar/Scim/User.hs

```diff
@@ -193,31 +194,35 @@ instance
 -- | Validate a raw SCIM user record and extract data that we care about. See also:
 -- 'ValidScimUser''.
 validateScimUser ::
-  forall m r.
-  (m ~ Scim.ScimHandler (Sem r)) =>
-  ( Member (Input Opts) r,
+  forall r.
+  ( Member SAMLUserStore r,
+    Member BrigAccess r,
+    Member (Input Opts) r,
     Member IdPConfigStore r
   ) =>
   Text ->
   -- | Used to decide what IdP to assign the user to
   ScimTokenInfo ->
   Scim.User ST.SparTag ->
-  m ST.ValidScimUser
+  Scim.ScimHandler (Sem r) ST.ValidScimUser
 validateScimUser errloc tokinfo user = do
   mIdpConfig <- tokenInfoToIdP tokinfo
   richInfoLimit <- lift $ inputs richInfoLimit
-  validateScimUser' errloc mIdpConfig richInfoLimit user
+  eitherUser <- lift $ runError $ validateScimUser' errloc mIdpConfig richInfoLimit user
+  case eitherUser of
+    Left err -> throwError err
+    Right validatedUser -> pure validatedUser
 
 tokenInfoToIdP :: Member IdPConfigStore r => ScimTokenInfo -> Scim.ScimHandler (Sem r) (Maybe IdP)
 tokenInfoToIdP ScimTokenInfo {stiIdP} =
   mapM (lift . IdPConfigStore.getConfig) stiIdP
 
 -- | Validate a handle (@userName@).
-validateHandle :: MonadError Scim.ScimError m => Text -> m Handle
+validateHandle :: Member (Error Scim.ScimError) r => Text -> Sem r Handle
```

Comment:
history: the old code made sense when you read `MonadError` as a generlization of `Either`, and want `validateHandle` to be a pure function for easier testing.  But polysemy (introduced as an afterthough) is just as testable.